figures/serializers.py

```diff
@@ -715,9 +715,12 @@ def get_courses(self, user):
         """
         This method is a hack until I figure out customizing DRF fields and/or
         related serializers to explicitly link models not linked via FK
+
         """
-        return LearnerCourseDetailsSerializer(
-            CourseEnrollment.objects.filter(user=user), many=True).data
+
+        course_enrollments = figures.sites.get_course_enrollments_for_site(
+            self.context.get('site')).filter(user=user)
```

Comment:
We now use DRF serializers `context` construct to provide the `site` model to the sites method

Reply:
what happens here if `site` isn't in the `context` dict?

Reply:
Good catch! This is a latent bug. The LearnerDetailsSerializer is only used in LearnerDetailsViewSet, so for the current state of the code, context always has a `site` key. I'll create a ticket to track and fix this.

Reply:
Good catch! This is a latent bug. Currently, LearnerDetailsSerializer is only called by LearnerDetailsViewSet. Created a ticket to track and address this: https://github.com/appsembler/figures/issues/87